### PR TITLE
🧪 test: Add table-driven tests for ParseNewsItem

### PR DESCRIPTION
**Pull request by arran4**
🎯 **What:** The untested `ParseNewsItem` function in `news_parser.go` lacked explicit coverage for its 1.0 format edge cases and individual fields.
📊 **Coverage:** Added a table-driven test suite (`TestParseNewsItem`) covering:
- Happy path (Basic Format 1.0)
- Missing headers
- Multiple occurrences of headers (Translators, Display-If-Installed, Display-If-Keyword, Display-If-Profile)
- Invalid date formats
- Invalid header formats (missing colon)
- Empty files and missing body
✨ **Result:** Improved test coverage and reliability for parsing Gentoo news items (GLEP 42), ensuring all edge cases are explicitly verified and potential regressions can be caught early.

---
*PR created automatically by Jules for task [12430964550442392407](https://jules.google.com/task/12430964550442392407) started by @arran4*

### CODE REVIEW

File: news_parser_test.go [hunk 1]
```diff
@@ -1,10 +1,159 @@
...
+			}
+			if !reflect.DeepEqual(got.DisplayIfKeyword, tt.expected.DisplayIfKeyword) {
+				t.Errorf("DisplayIfKeyword = %v, want %v", got.DisplayIfKeyword, tt.expected.DisplayIfKeyword)
+			}
+			if !reflect.DeepEqual(got.DisplayIfProfile, tt.expected.DisplayIfProfile) {
+				t.Errorf("DisplayIfProfile = %v, want %v", got.DisplayIfProfile, tt.expected.DisplayIfProfile)
+			}
+			if got.Body != tt.expected.Body {
+				t.Errorf("Body = %v, want %v", got.Body, tt.expected.Body)
+			}
```
[the remark this answers is not in the document]
Comment (arran4) [reply news_parser_test.go]: > medium
> 
> The manual field-by-field comparison is verbose and misses several fields (like `BodyAST`, `DirName`, and `FileName`), which reduces the effectiveness of the tests. Additionally, the comment on lines 120-122 is misleading as it suggests `BodyAST` might be checked, but the code skips it. For better maintainability and coverage, consider using `reflect.DeepEqual` on the entire struct. Also, when reporting string mismatches, using `%q` instead of `%v` is recommended to clearly identify whitespace or empty string issues.
> 
> ```go
> 			// Verify all fields of the NewsItem struct match the expected values.
> 			if !reflect.DeepEqual(got, tt.expected) {
> 				t.Errorf("ParseNewsItem() = %#v, want %#v", got, tt.expected)
> 			}
> ```

@jules
